plugins/parsers/logfmt/parser.go

```diff
@@ -0,0 +1,102 @@
+package logfmt
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/go-logfmt/logfmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+)
+
+var (
+	ErrNoMetric = fmt.Errorf("no metric in line")
+)
+
+// Parser decodes logfmt formatted messages into metrics.
+type Parser struct {
+	MetricName  string
+	DefaultTags map[string]string
+	Now         func() time.Time
+}
+
+// NewParser creates a parser.
+func NewParser(metricName string, defaultTags map[string]string) *Parser {
+	return &Parser{
+		MetricName:  metricName,
```

Comment:
What happens if this is not set?

Reply:
By default, doesn't telegraf use the plugin name of the input/processor that uses it?